recipes/pelegant/meta.yaml

```diff
@@ -0,0 +1,275 @@
+    - which sdds2mpl
+    - which sdds2plaindata
+    - which sdds2spreadsheet
+    - which sdds2stl
+    - which sdds2stream
+    - which sdds2tiff
+    - which sddsDiskUsage
+    - which sddsanalyticsignal
+    - which sddsarray2column
+    - which sddsbaseline
+    - which sddsbinarystring
+    - which sddsbreak
+    - which sddscast
+    - which sddschanges
+    - which sddscheck
+    - which sddscliptails
+    - which sddscollapse
+    - which sddscollect
+    - which sddscombine
+    - which sddscombinelogfiles
+    - which sddsconvert
+    - which sddsconvertlogonchange
+    - which sddsconvolve
+    - which sddscorrelate
+    - which sddsderef
+    - which sddsderiv
+    - which sddsdiff
+    - which sddsdigfilter
+    - which sddsdistest
+    - which sddsduplicate
+    - which sddsendian
+    - which sddsenvelope
+    - which sddseventhist
+    - which sddsexpand
+    - which sddsexpfit
+    - which sddsfdfilter
+    - which sddsfft
+    - which sddsfindin2dgrid
+    - which sddsgenericfit
+    - which sddsgfit
+    - which sddshist
+    - which sddsimageprofiles
+    - which sddsinsideboundaries
+    - which sddsinteg
+    - which sddsinterp
+    - which sddsinterpset
+    - which sddslogclient
+    - which sddslogserver
+    - which sddslorentzianfit
+    - which sddsmakedataset
+    - which sddsmatrix2column
+    - which sddsmatrixmult
+    - which sddsminterp
+    - which sddsmpfit
+    - which sddsmselect
+    - which sddsmultihist
+    - which sddsmxref
+    - which sddsnaff
+    - which sddsnormalize
+    - which sddsoutlier
+    - which sddspeakfind
+    - which sddspfit
+    - which sddsplot
+    - which sddspoly
+    - which sddsprintout
+    - which sddsprocess
+    - which sddsquery
+    - which sddsregroup
+    - which sddsremoveoffsets
+    - which sddsrespmatrixderivative
+    - which sddsrowstats
+    - which sddsrunstats
+    - which sddssampledist
+    - which sddsselect
+    - which sddsseparate
+    - which sddssequence
+    - which sddsshift
+    - which sddsshiftcor
+    - which sddssinefit
+    - which sddsslopes
+    - which sddssmooth
+    - which sddssnap2grid
+    - which sddssort
+    - which sddssortcolumn
+    - which sddssplit
+    - which sddsspotanalysis
+    - which sddstdrpeeling
+    - which sddstimeconvert
+    - which sddstranspose
+    - which sddsunwrap
+    - which sddsvslopes
+    - which sddsxref
+    - which sddszerofind
+    - which startFirefox
+    - which tcomp
+    - which tdms2sdds
+    - which tek2sdds
+    - which testlock
+    - which tiff2sdds
+    - which timeconvert
+    - which tmpname
+    - which token
+    - which wfm2sdds
+
+about:
+  home: https://www.aps.anl.gov/Accelerator-Operations-Physics/Software
+  license: EPICS
+  license_file:
+    - oag/apps/src/elegant/LICENSE
+    - fdlibm_license.txt
+    - libgd_license.txt
+    - meschach_license.txt
+    - pgapack_license.txt
+    - png_license.txt
+    - tiff_license.txt
+    - xlslib_license.txt
+    - zlib_license.txt
```

Comment:
Thanks for the detailed reponse. I'm just trying to do my due dilligence to prevent binary duplication on the channel. 

> The maintainers of Elegant do not specify compatibility but rather state "use this version of epics-base we provide". 

> `sdds` could perhaps be packaged separately, but it is likely not useful on its own to end-users

The above explanation seem like good reasons to not use existing epics packages and not split off SDDS into it's own package.

I still have reservations about the revending of certain popular general-use libraries with stable APIs. These include libpng, libgd, libtiff, and zlib.

My understanding is that these deps are only used directly by and revended by the SDDS component. The Make file for SDDS contains the following release notes:

```
# Revision 1.32  2007/04/17 17:24:33  soliday
# Updated to better detect and use png, gd, and tiff system libraries.
```
Which implies that upstream supports to linking to non-vended versions of these libraries.

I have compared the vended versions of these libraries in SDDS vs latest on the channel:

libpng 1.6.10 -> 1.6.43
libgd 2.1.0 -> 2.3.3
libtiff 42 -> 4.6.0 (not exactly sure how this project handles versions)
zlib 1.2.8 -> 1.3.1

Based on the version numbers, it seems reasonable that the vended versions and the latest versions on this channels are API compatible.